File: C++/443. String Compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int ans = 0,cnt = 1;
        int len = chars.size();
        char c = chars[0];
        for(int i = 1;i<len;i++)
        {
            if(chars[i]!=chars[i-1])
            {
                chars[ans++] = c;
                string tmp = to_string(cnt);
                if(cnt>1)
                {
                    for(int j=0;j<tmp.length();j++)
                        chars[ans++] = tmp[j];    
                }
                c = chars[i];
                cnt = 1;
            }
            else
                cnt++;
        }
        chars[ans++] = c;
        string tmp = to_string(cnt);
        if(cnt>1)
        {
            for(int j=0;j<tmp.length();j++)
                chars[ans++] = tmp[j];    
        }
        for(int i=ans;i<len;i++)
            chars.pop_back();
        return ans;
    }
};

File: Python/443. String Compression.py
class Solution:
    def compress(self, chars):
        """
        :type chars: List[str]
        :rtype: int
        """
        l, i = 0, 0
        while l < len(chars):
            c, cnt = chars[l], 1
            while l < len(chars) - 1 and chars[l] == chars[l+1]:
                cnt, l = cnt + 1, l + 1
            chars[i] = c
            if cnt > 1:
                tmp = str(cnt)
                chars[i + 1:i + 1 + len(tmp)] = tmp
                i += len(tmp)
            l, i = l + 1, i + 1
        return i